src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+        pixel_mask = torch.ones((y.size(0), y.size(1)), dtype=torch.long, device=self.device)
+        extend_image_masks = self.text_transformer.get_extended_attention_mask(
+            pixel_mask, pixel_mask.size(), self.device
+        )
+
+        x1 = self.cross_modal_text_layers[0](x, y, extend_text_masks, extend_image_masks)[0]
+        y1 = self.cross_modal_image_layers[0](y, x, extend_image_masks, extend_text_masks)[0]
+
+        link_layer_index = 0
+
+        # link tower fusion
+        for i in range(split_index, len(self.text_transformer.encoder.layer)):
+            text_embeds = self.text_transformer.encoder.layer[i](text_embeds, extend_text_masks)[0]
+            image_embeds = self.vit_model.visual.transformer.resblocks[i](image_embeds).type(self.vit_model.dtype)
+            image_embeds_ = (
+                self.cross_modal_image_transform(self.vit_model.visual.forward_post(image_embeds))
+                + image_token_type_embeddings
+            )
+
+            text_link_tower = self.cross_modal_text_link_tower[link_layer_index]
+            image_link_tower = self.cross_modal_image_link_tower[link_layer_index]
+
+            x1_ = text_link_tower(
+                self.cross_modal_text_transform(text_embeds) + text_token_type_embeddings, x1, extend_text_masks
+            )
+            if irtr_len > 0:
+                y1_ = image_link_tower(
+                    image_embeds_.unsqueeze(1).expand(_bs, irtr_len, _L, _D).reshape(-1, _L, _D),
+                    y1,
+                    extend_image_masks,
+                )
+            else:
+                y1_ = image_link_tower(image_embeds_, y1, extend_image_masks)
+
+            x1 = self.cross_modal_text_layers[link_layer_index + 1](x1_, y1_, extend_text_masks, extend_image_masks)[0]
+            y1 = self.cross_modal_image_layers[link_layer_index + 1](y1_, x1_, extend_image_masks, extend_text_masks)[
+                0
+            ]
+
+            link_layer_index += 1
+
+        text_feats, image_feats = x1, y1
+        cls_feats = self.get_cls_feats(text_feats, image_feats)
+
+        if not return_dict:
+            return tuple(v for v in [text_feats, image_feats, cls_feats] if v is not None)
+
+        return BridgeTowerModelOutput(
+            text_feats=text_feats,
+            image_feats=image_feats,
+            pooler_output=cls_feats,
+        )
+
+    def get_cls_feats(self, text_feats, image_feats):
+        cls_feats_text = self.cross_modal_text_pooler(text_feats)
+        if self.is_clip:
+            cls_feats_image = self.cross_modal_image_pooler(image_feats)
+        else:
+            avg_image_feats = self.avgpool(image_feats.transpose(1, 2)).view(image_feats.size(0), 1, -1)
+            cls_feats_image = self.cross_modal_image_pooler(avg_image_feats)
+        return torch.cat([cls_feats_text, cls_feats_image], dim=-1)
+
+
+class LinkTower(nn.Module):
+    def __init__(self, config, tokenizer_config):
```

Comment:
```suggestion
    def __init__(self, config):
```
the tokenizer config can be added as an attribute to the main config

Reply:
Handled in the updated commits.